doc/BGP/BGP-aggregate-address.md

```diff
@@ -0,0 +1,211 @@
+                    description "Aggregate address to be advertised";
+                }
+
+                leaf bbr-required {
+                    type boolean;
+                    description "Set if BBR is required for generating aggregate address";
+                }
+
+                leaf summary-only {
+                    type boolean;
+                    description "Only advertise the summary of aggregate address";
+                }
+
+                leaf route-map {
+                    type string;
+                    description "Attribute-map to be applied to the aggregate address";
+                }
+
+                leaf as-set {
+                     type boolean;
+                     description "Set if include the AS set when advertising the aggregated address";
+                }
+            }
+        }
+    }
+}
+```
+
+#### Config DB Sample
+```json
+{
+    ...
+    "BGP_AGGREGATE_ADDRESS": {
+        "192.168.0.0/24": {
+            "bbr-required": "true",
+            "summary-only": "false"
+        },
+        "fc00::/63": {
+            "bbr-required": "true",
+            "summary-only": "true"
+            "route-map": "AGG_V6",
+            "as-set": "true"
+        }
+    }
+    ...
+}
+```
+
+#### Parameters
+We have parameters to control the behavior of aggregate address in bgp container.
+
+##### BBR Required
+It's a boolean to indicate whether only generate aggregated address when BBR feature is enabled.
+
+If it's true and the BBR feature is not enabled, aggregated address won't generated.
+
+##### Summary Only
+It's a boolean to indicate whether only advertise aggregate address only.
+
+If it's true, then details routes will be suppressed and only aggregated address will be advertised.
+
+##### AS Set
+It's a boolean to indicate whether add a as set of details routes in as path when advertising aggregated address.
+
+If it's true, then when advertising aggregated address there will be a as set of detail routes appended at as path.
+
+
+##### Route Map
+It's a string to indicate which route map to apply to aggregated address.
+
+It should be a name of route map.
+
+### State DB Extension
+For every aggregated address, we track its state in state DB, it has two states active and inactive. Active state means the address is configurated in the bgp container, while inactive state means isn't.
+
+#### State DB sample:
+```json
+{
+    ...
+    "BGP_AGGREGATE_ADDRESS": {
+        "192.168.0.0/24": {
+            "state": "inactive"
+        },
+        "fc00::/63": {
+            "state": "active"
+        }
+    }
+    ...
+}
+```
+
+#### State Transition Diagram
+
+<p align=center>
+<img src="img/aggregate_address_state_transition.png" alt="state-transition">
+</p>
+
+
+### Bgp Container Behavior
+The bgp container will subscribe the keys `BGP_AGGREGATE_ADDRESS` and `BGP_BBR` in config DB. The possible events and related process are:
+1. Add address in config DB:
```

Comment:
Please add UML seq diagrams for below operations
Involved components:
config-db
state-db
bgpcfgd
bgpaggre-cfg...